chat/consumers.py

```diff
@@ -0,0 +1,43 @@
+import json
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+class ChatConsumer(AsyncWebsocketConsumer):
+    """
+    WebSocket consumer for handling real-time chat messages in specific rooms.
+    """
+
+    async def connect(self):
+        """
+        Handles a new WebSocket connection.
+        Joins the corresponding room group based on the room name.
+        """
+        self.room_name = self.scope["url_route"]["kwargs"]["room_name"]
+        self.room_group_name = f"chat_{self.room_name}"
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
+        await self.accept()
+
+    async def disconnect(self, close_code):
+        """
+        Handles WebSocket disconnection.
+        Removes the connection from the room group.
+        """
+        await self.channel_layer.group_discard(self.room_group_name, self.channel_name)
+
+    async def receive(self, text_data):
+        """
+        Handles incoming messages from the WebSocket client.
+        Broadcasts the message to all members of the room group.
+        """
+        text_data_json = json.loads(text_data)
+        message = text_data_json["message"]
```

Comment:
Accessing message directly (message = text_data_json["message"]) may throw a KeyError if "message" is not present. Use .get("message") and handle None or missing key gracefully.